src/core/lib/security/security_connector/local/local_security_connector.cc

```diff
@@ -70,39 +77,19 @@ void local_check_peer(tsi_peer peer, grpc_endpoint* ep,
                       grpc_core::RefCountedPtr<grpc_auth_context>* auth_context,
                       grpc_closure* on_peer_checked,
                       grpc_local_connect_type type) {
-  int fd = grpc_endpoint_get_fd(ep);
-  grpc_resolved_address resolved_addr;
-  memset(&resolved_addr, 0, sizeof(resolved_addr));
-  resolved_addr.len = GRPC_MAX_SOCKADDR_SIZE;
   bool is_endpoint_local = false;
-  if (getsockname(fd, reinterpret_cast<grpc_sockaddr*>(resolved_addr.addr),
-                  &resolved_addr.len) == 0) {
-    grpc_resolved_address addr_normalized;
-    grpc_resolved_address* addr =
-        grpc_sockaddr_is_v4mapped(&resolved_addr, &addr_normalized)
-            ? &addr_normalized
-            : &resolved_addr;
-    grpc_sockaddr* sock_addr = reinterpret_cast<grpc_sockaddr*>(&addr->addr);
-    // UDS
-    if (type == UDS && grpc_is_unix_socket(addr)) {
+
+  char* peer_address = ep->vtable->get_peer(ep);
+  if (type == UDS && strncmp(k_unix, peer_address, n_unix) == 0) {
```

Comment:
Instead of doing string parsing here, I suggest converting the peer address URI into a `grpc_resolved_address` using something like this:

https://github.com/grpc/grpc/blob/eb0b2067321502d1376dd7ba8d5d345865528601/src/core/ext/filters/client_channel/subchannel.cc#L912

Then you can basically use the existing code to check the value of the `grpc_resolved_address`.

Reply:
Copied the method here to parse URI correctly.